### PR TITLE
Added `b pm-info module` (alias pmi) command

### PR DESCRIPTION
**Pull request by alanmels**
Implementation of https://github.com/backdrop-contrib/b/issues/55

### CODE REVIEW

**Comment by bradbulger** [comment]
Runs great.

I don't know the module code well enough to comment on the implementation too much. I think IF ELSE structures are supposed to put the "else" on a new line? For what that's worth.

File: commands/PM/modules.b.inc [hunk 1]
```diff
@@ -438,3 +450,108 @@ function _uninstall_project($project) {
...
+ *
+ * @param array $arguments
+ * @param array $options
+ */
+function modules_b_info_callback($arguments, $options) {
+  if (!isset($arguments['modules'])) {
+    b_set_message('Please provide project name(s)', 'error');
+    return;
+  }
+  foreach($arguments['modules'] as $project ) {
```
Comment (klonos) [inline commands/PM/modules.b.inc]:
```suggestion
  foreach($arguments['modules'] as $project) {
```

**Review by AlexShapka — APPROVED** [review]

**Comment by AlexShapka** [comment]
> Runs great.
> 
> I don't know the module code well enough to comment on the implementation too much. I think IF ELSE structures are supposed to put the "else" on a new line? For what that's worth.

Thanks for the suggestion. I've changed the conditions in different way.

File: commands/PM/modules.b.inc [hunk 2]
```diff
@@ -438,3 +450,108 @@ function _uninstall_project($project) {
...
+
+/**
+ * Internal function to provide detailed information about module or theme.
+ *
+ * @param string $project
+ *  The machine name of the project about which info is required.
+ */
+function _info_project($project) {
+  $query = db_select('system', 's')
+      ->fields('s');
```
Comment (klonos) [inline commands/PM/modules.b.inc:478]: Not entirely sure whether this way of line wrapping is correct according to coding standards; not sure about the indentation either.

File: commands/PM/modules.b.inc [hunk 3]
```diff
@@ -438,3 +450,108 @@ function _uninstall_project($project) {
...
+  $query->condition('name', $project);
+  $query->condition('type', 'module');
+
+  $module = $query->execute()->fetchAssoc();
+  if (!$module) {
+    b_set_message(bt("Module !name not found", array('!name' => $project)), 'error');
+  } else {
```
Comment (klonos) [inline commands/PM/modules.b.inc:485]:
```suggestion
  }
  else {
```

File: commands/PM/modules.b.inc [hunk 4]
```diff
@@ -438,3 +450,108 @@ function _uninstall_project($project) {
...
+      format_date($info['timestamp'], 'custom', 'Y-m-d'),
+    );
+    $rows[] = array(
+      array('data' => bt('PHP'), '#color' => 'bold_blue'),
+      $info['php'],
+    );
+
+    if (!empty($info['files'])) {
+      $files = implode(', ', $info['files']);
+    } else {
```
Comment (klonos) [inline commands/PM/modules.b.inc]:
```suggestion
    }
    else {
```

File: commands/PM/modules.b.inc [hunk 5]
```diff
@@ -438,3 +450,108 @@ function _uninstall_project($project) {
...
+      $files = 'none';
+    }
+    $rows[] = array(
+      array('data' => bt('Files'), '#color' => 'bold_blue'),
+      $files,
+    );
+
+    if (!empty($info['dependencies'])) {
+      $dependencies = implode(', ', $info['dependencies']);
+    } else {
```
Comment (klonos) [inline commands/PM/modules.b.inc]:
```suggestion
    }
    else {
```

**Comment by serundeputy** [comment]
@AlexShapka I appreciate your enthusiasm here!

Please don't commit code to `backdrop-contrib` modules that you are not the maintainer of.

I'm sure this feature works great and is a great improvement to `b`, but we have not had a chance to test it yet and make sure that it works as advertised and does not break other features.

I know the github permissions allow you to commit code to other projects, but as a matter of respect to the other maintainers we do not commit code to there projects, i.e. let the maintainers maintain.

**Comment by AlexShapka** [comment]
@serundeputy I do apologize. Will never happen again. It was in fact my first PR merge and I got somehow confused.

File: commands/PM/modules.b.inc (hunk 2 above)
Comment (ghost) [reply commands/PM/modules.b.inc:478]: It's ok, we have coding standards checks in (`b`) core now :grin:

**Pull request merged**